javalib/src/main/scala/java/util/Date.scala

```diff
@@ -120,24 +131,36 @@ class Date private (private val date: js.Date) extends Object
 
   @Deprecated
   def toLocaleString(): String = {
+    val date = asDate()
     "" + date.getDate().toInt + "-" + Months(date.getMonth().toInt) + "-" +
       date.getFullYear().toInt + "-" + pad0(date.getHours().toInt) + ":" +
       pad0(date.getMinutes().toInt) + ":" + pad0(date.getSeconds().toInt)
   }
 
   override def toString(): String = {
-    val offset = -date.getTimezoneOffset().toInt
-    val sign = if (offset < 0) "-" else "+"
-    val hours = pad0(Math.abs(offset) / 60)
-    val mins = pad0(Math.abs(offset) % 60)
-    Days(date.getDay().toInt) + " "+ Months(date.getMonth().toInt) + " " +
-      pad0(date.getDate().toInt) + " " + pad0(date.getHours().toInt) + ":" +
-      pad0(date.getMinutes().toInt) + ":" + pad0(date.getSeconds().toInt) +
-      " GMT" + " " + date.getFullYear().toInt
+    if (isSafeJSDate()) {
+      val date = asDate()
+      val offset = -date.getTimezoneOffset().toInt
+      val sign = if (offset < 0) "-" else "+"
+      val hours = pad0(Math.abs(offset) / 60)
+      val mins = pad0(Math.abs(offset) % 60)
+      Days(date.getDay().toInt) + " "+ Months(date.getMonth().toInt) + " " +
+        pad0(date.getDate().toInt) + " " + pad0(date.getHours().toInt) + ":" +
+        pad0(date.getMinutes().toInt) + ":" + pad0(date.getSeconds().toInt) +
+        " GMT" + " " + date.getFullYear().toInt
+    } else {
+      f"java.util.Date($millis)"
+    }
   }
+
+  @inline
+  private def isSafeJSDate(): Boolean =
+    -MaxMillis <= millis && millis <= MaxMillis
 }
 
 object Date {
+  private final val MaxMillis = 8640000000000000L
```

Comment:
This constant deserves a comment ;) Perhaps a link to [`TimeClip()`](https://tc39.es/ecma262/#sec-timeclip) in the ES spec?